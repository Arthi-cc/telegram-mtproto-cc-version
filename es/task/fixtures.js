export var initFlags = ({
  api = false,
  inner = false,
  container = false,
  incoming = true,
  methodResult = false,
  body = false,
  error = false
}) => ({
  api,
  inner,
  container,
  incoming,
  methodResult,
  body,
  error
});

export function isApiObject(obj) {
  return typeof obj === 'object' && obj != null && typeof obj._ === 'string';
}
//# sourceMappingURL=fixtures.js.map