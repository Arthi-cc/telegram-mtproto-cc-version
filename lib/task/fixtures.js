'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.isApiObject = isApiObject;
var initFlags = exports.initFlags = ({
  api = false,
  inner = false,
  container = false,
  incoming = true,
  methodResult = false,
  body = false,
  error = false
}) => ({
  api,
  inner,
  container,
  incoming,
  methodResult,
  body,
  error
});

function isApiObject(obj) {
  return typeof obj === 'object' && obj != null && typeof obj._ === 'string';
}
//# sourceMappingURL=fixtures.js.map