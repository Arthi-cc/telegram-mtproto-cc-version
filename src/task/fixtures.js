//@flow

export const initFlags = ({
  api         = false,
  inner       = false,
  container   = false,
  incoming    = true,
  methodResult= false,
  body        = false,
  error       = false,
}: {
  api?: boolean,
  inner?: boolean,
  container?: boolean,
  incoming?: boolean,
  methodResult?: boolean,
  body?: boolean,
  error?: boolean,
}): {
  api: boolean,
  inner: boolean,
  container: boolean,
  incoming: boolean,
  methodResult: boolean,
  body: boolean,
  error: boolean,
} => ({
  api,
  inner,
  container,
  incoming,
  methodResult,
  body,
  error,
})

export function isApiObject(obj: mixed): boolean %checks {
  return (
    typeof obj === 'object'
    && obj != null
    && typeof obj._ === 'string'
  )
}
