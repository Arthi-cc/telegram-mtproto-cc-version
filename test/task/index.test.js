
import { api } from '../mtproto-config'
import { freezeTime } from './fixtures'

let client

beforeAll(() => {
  jest.mock('../../src/config-provider/provider')
  const { MTProto } = require('../../src')
  client = MTProto({
    server: {
      dev     : true,
      webogram: true
    },
    api
  })
  client.emit('deactivate')
})


afterAll(() => {
  jest.unmock('../../src/config-provider/provider')
})

test('binary parsing', async() => {
  const parser = require('../../src/service/chain').default
  const binaryParseData = require('./example-normalize').default
  await expect(parser(binaryParseData)).resolves.toMatchSnapshot()
})


test('normalize', async() => {
  const normalize = require('../../src/task').default
  const { rawData, authKey, serverSalt, sessionID, getMsgById } = require('./example-normalize')
  const { default: result, TIMESTAMP } = require('./example-target')
  const unfreezeTime = freezeTime(TIMESTAMP)

  const thread = client.api.networkSetter(2, authKey, serverSalt) //TODO networkSetter was removed!
  thread.sessionID = sessionID
  thread.getMsgById = getMsgById
  const data = { ...rawData, thread }
  await expect(normalize(data)).resolves.toEqual(result)
  unfreezeTime()
})
