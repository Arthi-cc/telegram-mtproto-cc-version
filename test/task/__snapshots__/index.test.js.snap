// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`binary parsing 1`] = `
Object {
  "messageID": "6451324666894843905",
  "response": Object {
    "_": "msg_container",
    "messages": Array [
      Object {
        "_": "message",
        "body": Object {
          "_": "new_session_created",
          "first_msg_id": "6451324639565242596",
          "server_salt": "10054112889999250563",
          "unique_id": "12719723951254540931",
        },
        "bytes": 28,
        "msg_id": "6451324664439027713",
        "seqno": 1,
      },
      Object {
        "_": "message",
        "body": Object {
          "_": "rpc_result",
          "req_msg_id": "6451324639565242596",
          "result": Object {
            "_": "auth.sentCode",
            "flags": 3,
            "next_type": Object {
              "_": "auth.codeTypeSms",
            },
            "phone_code_hash": "e49a9dab2598a3db61",
            "phone_registered": true,
            "type": Object {
              "_": "auth.sentCodeTypeApp",
              "length": 5,
            },
          },
        },
        "bytes": 52,
        "msg_id": "6451324664531753985",
        "seqno": 3,
      },
    ],
  },
  "seqNo": 4,
  "sessionID": Uint8Array [
    104,
    253,
    6,
    5,
    166,
    213,
    190,
    141,
  ],
}
`;
